enterprise_access/apps/subsidy_access_policy/subsidy_api.py

```diff
@@ -146,8 +146,15 @@ def get_redemptions_by_content_and_policy_for_learner(policies, lms_user_id):
             content_key = redemption['content_key']
             subsidy_access_policy_uuid = redemption['subsidy_access_policy_uuid']
 
-            if subsidy_access_policy_uuid in policies_with_subsidy:
-                result[content_key][subsidy_access_policy_uuid].append(redemption)
+            matching_policies = [
+                policy for policy in policies_with_subsidy
+                if str(policy.uuid) == subsidy_access_policy_uuid
+            ]
+            # We can assume there's at most one matching policy because the ``policies`` arg passed into this function
+            # should not contain duplicates.
+            matching_policy = matching_policies[0] if matching_policies else None
+            if matching_policy:
+                result[content_key][matching_policy].append(redemption)
```

Comment:
I slightly refactored this function to return different mapping:

* old: dict(content_key -> dict(policy UUID -> list of redemptions)
* new: dict(content_key -> dict(**policy instance** -> list of redemptions)

The purpose is to avoid another DB call in downstream code to fetch the policy instance.  We already pulled it into memory here in this function, so just don't discard it yet.